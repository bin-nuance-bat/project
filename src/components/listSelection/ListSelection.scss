@import '../../utils/constants.scss';

$icon-distance: 56px;
$icon-width: 48px;
$item-text-distance: 24px;

.itemicon {
	width: $icon-width;
	height: $icon-width;
	position: absolute;
	left: $icon-distance;
}

.listselection {
	padding-top: 32px;
	height: $icon-width;
	background-color: $white;
	position: relative;
	height: 100%;
}

.listtext {
	color: $text-blue;
	font-size: 30px;
	position: relative;
	text-align: left;
}

.listitem {
	padding-bottom: 94px;
	vertical-align: middle;
	height: $icon-width;
}

.itemtext {
	padding-left: $icon-distance + $icon-width + $item-text-distance;
}

.grouptext {
	padding-left: 32px;
}
