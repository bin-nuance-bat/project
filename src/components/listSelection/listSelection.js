import React from 'react';
import _ from 'lodash';
import './ListSelection.css';
import line from './line.svg';

const compare = (a, b) => {
	if (a[0] > b[0]) return 1;
	if (a[0] < b[0]) return -1;
	return 0;
};

const getGroupedItems = items => {
	const groupList = _.groupBy(items, item => item.name[0]);
	return Object.entries(groupList).sort(compare);
};

const ListSelection = props => {
	const groupedItems = getGroupedItems(props.items);
	return (
		<div className="listselection">
			{groupedItems.map(([group, groupItems]) => (
				<div className="listgroup" key={group}>
					<p className="listtext grouptext">{group.toUpperCase()}</p>
					<img src={line} alt="" />
					{groupItems.map(item => (
						<div
							className="listitem"
							key={item.id}
							name={item.id}
							onClick={e =>
								props.onClick(e.target.getAttribute('name'))
							}>
							<img
								name={item.id}
								className="itemicon"
								src={item.image}
								alt=""
							/>
							<p name={item.id} className="listtext itemtext">
								{item.name}
							</p>
						</div>
					))}
				</div>
			))}
		</div>
	);
};

export default ListSelection;
