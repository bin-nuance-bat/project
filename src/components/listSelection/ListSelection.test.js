import React from 'react';
import {shallow, configure} from 'enzyme';
import ListSelection from './ListSelection';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

it('Gnerates the list correctly', () => {
	const props = {
		onClick: jest.fn(),
		items: [
			{name: 'foo', id: 'sample-foo', image: ''},
			{name: 'bar', id: 'sample-bar', image: ''},
			{name: 'foobar', id: 'sample-foobar', image: ''}
		]
	};
	const wrapper = shallow(<ListSelection {...props} />);
	expect(wrapper).toMatchSnapshot();
});

it('Calls the required function', () => {
	const props = {
		onClick: jest.fn(),
		items: [
			{name: 'foo', id: 'sample-foo', image: ''},
			{name: 'bar', id: 'sample-bar', image: ''},
			{name: 'foobar', id: 'sample-foobar', image: ''}
		]
	};

	const wrapper = shallow(<ListSelection {...props} />);
	wrapper
		.find('[name="foo"]')
		.at(0)
		.simulate('click');
	expect(props.onClick).toHaveBeenCalledWith('sample-foo');
});
