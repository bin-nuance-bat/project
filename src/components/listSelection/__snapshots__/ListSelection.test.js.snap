// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gnerates the list correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListSelection
    items={
        Array [
            Object {
              "id": "sample-foo",
              "image": "",
              "name": "foo",
            },
            Object {
              "id": "sample-bar",
              "image": "",
              "name": "bar",
            },
            Object {
              "id": "sample-foobar",
              "image": "",
              "name": "foobar",
            },
          ]
    }
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="listgroup"
>
          <p
                    className="listtext grouptext"
          >
                    B
          </p>
          <img
                    alt=""
                    src="line.svg"
          />
          <div
                    className="listitem"
                    name="sample-bar"
                    onClick={[Function]}
          >
                    <img
                              alt=""
                              className="itemicon"
                              name="sample-bar"
                              src=""
                    />
                    <p
                              className="listtext itemtext"
                              name="sample-bar"
                    >
                              bar
                    </p>
          </div>
</div>,
        <div
          className="listgroup"
>
          <p
                    className="listtext grouptext"
          >
                    F
          </p>
          <img
                    alt=""
                    src="line.svg"
          />
          <div
                    className="listitem"
                    name="sample-foo"
                    onClick={[Function]}
          >
                    <img
                              alt=""
                              className="itemicon"
                              name="sample-foo"
                              src=""
                    />
                    <p
                              className="listtext itemtext"
                              name="sample-foo"
                    >
                              foo
                    </p>
          </div>
          <div
                    className="listitem"
                    name="sample-foobar"
                    onClick={[Function]}
          >
                    <img
                              alt=""
                              className="itemicon"
                              name="sample-foobar"
                              src=""
                    />
                    <p
                              className="listtext itemtext"
                              name="sample-foobar"
                    >
                              foobar
                    </p>
          </div>
</div>,
      ],
      "className": "listselection",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "b",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="listtext grouptext"
>
              B
</p>,
            <img
              alt=""
              src="line.svg"
/>,
            Array [
              <div
                className="listitem"
                name="sample-bar"
                onClick={[Function]}
>
                <img
                                alt=""
                                className="itemicon"
                                name="sample-bar"
                                src=""
                />
                <p
                                className="listtext itemtext"
                                name="sample-bar"
                >
                                bar
                </p>
</div>,
            ],
          ],
          "className": "listgroup",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "B",
              "className": "listtext grouptext",
            },
            "ref": null,
            "rendered": "B",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": "line.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": "sample-bar",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt=""
                  className="itemicon"
                  name="sample-bar"
                  src=""
/>,
                <p
                  className="listtext itemtext"
                  name="sample-bar"
>
                  bar
</p>,
              ],
              "className": "listitem",
              "name": "sample-bar",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "itemicon",
                  "name": "sample-bar",
                  "src": "",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "bar",
                  "className": "listtext itemtext",
                  "name": "sample-bar",
                },
                "ref": null,
                "rendered": "bar",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "f",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="listtext grouptext"
>
              F
</p>,
            <img
              alt=""
              src="line.svg"
/>,
            Array [
              <div
                className="listitem"
                name="sample-foo"
                onClick={[Function]}
>
                <img
                                alt=""
                                className="itemicon"
                                name="sample-foo"
                                src=""
                />
                <p
                                className="listtext itemtext"
                                name="sample-foo"
                >
                                foo
                </p>
</div>,
              <div
                className="listitem"
                name="sample-foobar"
                onClick={[Function]}
>
                <img
                                alt=""
                                className="itemicon"
                                name="sample-foobar"
                                src=""
                />
                <p
                                className="listtext itemtext"
                                name="sample-foobar"
                >
                                foobar
                </p>
</div>,
            ],
          ],
          "className": "listgroup",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "F",
              "className": "listtext grouptext",
            },
            "ref": null,
            "rendered": "F",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": "line.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": "sample-foo",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt=""
                  className="itemicon"
                  name="sample-foo"
                  src=""
/>,
                <p
                  className="listtext itemtext"
                  name="sample-foo"
>
                  foo
</p>,
              ],
              "className": "listitem",
              "name": "sample-foo",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "itemicon",
                  "name": "sample-foo",
                  "src": "",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "foo",
                  "className": "listtext itemtext",
                  "name": "sample-foo",
                },
                "ref": null,
                "rendered": "foo",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "sample-foobar",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt=""
                  className="itemicon"
                  name="sample-foobar"
                  src=""
/>,
                <p
                  className="listtext itemtext"
                  name="sample-foobar"
>
                  foobar
</p>,
              ],
              "className": "listitem",
              "name": "sample-foobar",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "itemicon",
                  "name": "sample-foobar",
                  "src": "",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "foobar",
                  "className": "listtext itemtext",
                  "name": "sample-foobar",
                },
                "ref": null,
                "rendered": "foobar",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="listgroup"
>
            <p
                        className="listtext grouptext"
            >
                        B
            </p>
            <img
                        alt=""
                        src="line.svg"
            />
            <div
                        className="listitem"
                        name="sample-bar"
                        onClick={[Function]}
            >
                        <img
                                    alt=""
                                    className="itemicon"
                                    name="sample-bar"
                                    src=""
                        />
                        <p
                                    className="listtext itemtext"
                                    name="sample-bar"
                        >
                                    bar
                        </p>
            </div>
</div>,
          <div
            className="listgroup"
>
            <p
                        className="listtext grouptext"
            >
                        F
            </p>
            <img
                        alt=""
                        src="line.svg"
            />
            <div
                        className="listitem"
                        name="sample-foo"
                        onClick={[Function]}
            >
                        <img
                                    alt=""
                                    className="itemicon"
                                    name="sample-foo"
                                    src=""
                        />
                        <p
                                    className="listtext itemtext"
                                    name="sample-foo"
                        >
                                    foo
                        </p>
            </div>
            <div
                        className="listitem"
                        name="sample-foobar"
                        onClick={[Function]}
            >
                        <img
                                    alt=""
                                    className="itemicon"
                                    name="sample-foobar"
                                    src=""
                        />
                        <p
                                    className="listtext itemtext"
                                    name="sample-foobar"
                        >
                                    foobar
                        </p>
            </div>
</div>,
        ],
        "className": "listselection",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "b",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="listtext grouptext"
>
                B
</p>,
              <img
                alt=""
                src="line.svg"
/>,
              Array [
                <div
                  className="listitem"
                  name="sample-bar"
                  onClick={[Function]}
>
                  <img
                                    alt=""
                                    className="itemicon"
                                    name="sample-bar"
                                    src=""
                  />
                  <p
                                    className="listtext itemtext"
                                    name="sample-bar"
                  >
                                    bar
                  </p>
</div>,
              ],
            ],
            "className": "listgroup",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "B",
                "className": "listtext grouptext",
              },
              "ref": null,
              "rendered": "B",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "src": "line.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": "sample-bar",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt=""
                    className="itemicon"
                    name="sample-bar"
                    src=""
/>,
                  <p
                    className="listtext itemtext"
                    name="sample-bar"
>
                    bar
</p>,
                ],
                "className": "listitem",
                "name": "sample-bar",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "itemicon",
                    "name": "sample-bar",
                    "src": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "bar",
                    "className": "listtext itemtext",
                    "name": "sample-bar",
                  },
                  "ref": null,
                  "rendered": "bar",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "f",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="listtext grouptext"
>
                F
</p>,
              <img
                alt=""
                src="line.svg"
/>,
              Array [
                <div
                  className="listitem"
                  name="sample-foo"
                  onClick={[Function]}
>
                  <img
                                    alt=""
                                    className="itemicon"
                                    name="sample-foo"
                                    src=""
                  />
                  <p
                                    className="listtext itemtext"
                                    name="sample-foo"
                  >
                                    foo
                  </p>
</div>,
                <div
                  className="listitem"
                  name="sample-foobar"
                  onClick={[Function]}
>
                  <img
                                    alt=""
                                    className="itemicon"
                                    name="sample-foobar"
                                    src=""
                  />
                  <p
                                    className="listtext itemtext"
                                    name="sample-foobar"
                  >
                                    foobar
                  </p>
</div>,
              ],
            ],
            "className": "listgroup",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "F",
                "className": "listtext grouptext",
              },
              "ref": null,
              "rendered": "F",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "src": "line.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": "sample-foo",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt=""
                    className="itemicon"
                    name="sample-foo"
                    src=""
/>,
                  <p
                    className="listtext itemtext"
                    name="sample-foo"
>
                    foo
</p>,
                ],
                "className": "listitem",
                "name": "sample-foo",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "itemicon",
                    "name": "sample-foo",
                    "src": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "foo",
                    "className": "listtext itemtext",
                    "name": "sample-foo",
                  },
                  "ref": null,
                  "rendered": "foo",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "sample-foobar",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt=""
                    className="itemicon"
                    name="sample-foobar"
                    src=""
/>,
                  <p
                    className="listtext itemtext"
                    name="sample-foobar"
>
                    foobar
</p>,
                ],
                "className": "listitem",
                "name": "sample-foobar",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "itemicon",
                    "name": "sample-foobar",
                    "src": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "foobar",
                    "className": "listtext itemtext",
                    "name": "sample-foobar",
                  },
                  "ref": null,
                  "rendered": "foobar",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
